main.go

```diff
@@ -106,22 +132,154 @@ func main() {
 			labelValue:             labelValue,
 		}
 		c := newController(klient, logger, opt)
+		c.registerMetrics(reg)
 		done := make(chan struct{})
 
 		g.Add(func() error {
 			return c.run(done)
-		}, func(err error) {
+		}, func(_ error) {
 			level.Info(logger).Log("msg", "shutting down controller")
 			close(done)
 		})
 	}
-	level.Info(logger).Log("msg", "starting the controller")
+	{
+		router := http.NewServeMux()
+		router.Handle("/metrics", promhttp.InstrumentMetricHandler(reg, promhttp.HandlerFor(reg, promhttp.HandlerOpts{})))
+		srv := &http.Server{Addr: config.InternalAddr, Handler: router}
 
+		g.Add(func() error {
+			return srv.ListenAndServe()
+		}, func(err error) {
+			if err == http.ErrServerClosed {
+				level.Warn(logger).Log("msg", "internal server closed unexpectedly")
+				return
+			}
+			level.Info(logger).Log("msg", "shutting down internal server")
+			ctx, _ := context.WithTimeout(context.Background(), internalServerShutdownTimeout)
+			if err := srv.Shutdown(ctx); err != nil {
+				stdlog.Fatal(err)
+			}
+		})
+	}
+
+	level.Info(logger).Log("msg", "starting the controller")
 	if err := g.Run(); err != nil {
 		stdlog.Fatal(err)
 	}
 }
 
+type prometheusReflectorMetrics struct {
+	listsMetric               prometheus.Counter
+	listDurationMetric        prometheus.Summary
+	itemsInListMetric         prometheus.Summary
+	watchesMetric             prometheus.Counter
+	shortWatchesMetric        prometheus.Counter
+	watchDurationMetric       prometheus.Summary
+	itemsInWatchMetric        prometheus.Summary
+	lastResourceVersionMetric prometheus.Gauge
+}
+
+func newReflactorMetrics(reg *prometheus.Registry) prometheusReflectorMetrics {
```

Comment:
Is this the intended spelling, or should it be `newReflectorMetrics`?

Reply:
Definitely not intended 🙈 Thanks